multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/FileUploadThreadPoolConfiguration.java

```diff
@@ -0,0 +1,158 @@
+package org.cloudfoundry.multiapps.controller.process.util;
+
+import java.util.Comparator;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.PriorityBlockingQueue;
+import java.util.concurrent.RunnableFuture;
+import java.util.concurrent.SynchronousQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import javax.inject.Inject;
+
+import org.cloudfoundry.multiapps.controller.core.util.ApplicationConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class FileUploadThreadPoolConfiguration {
+
+    private final ApplicationConfiguration applicationConfiguration;
+
+    @Inject
+    public FileUploadThreadPoolConfiguration(ApplicationConfiguration applicationConfiguration) {
+        this.applicationConfiguration = applicationConfiguration;
+    }
+
+    public static class PriorityFuture<T> implements RunnableFuture<T> {
+
+        private final RunnableFuture<T> src;
+        private final int priority;
+
+        public PriorityFuture(RunnableFuture<T> other, int priority) {
+            this.src = other;
+            this.priority = priority;
+        }
+
+        public int getPriority() {
+            return priority;
+        }
+
+        @Override
+        public boolean cancel(boolean mayInterruptIfRunning) {
+            return src.cancel(mayInterruptIfRunning);
+        }
+
+        @Override
+        public boolean isCancelled() {
+            return src.isCancelled();
+        }
+
+        @Override
+        public boolean isDone() {
+            return src.isDone();
+        }
+
+        @Override
+        public T get() throws InterruptedException, ExecutionException {
+            return src.get();
+        }
+
+        @Override
+        public T get(long timeout, TimeUnit unit) throws InterruptedException, ExecutionException, TimeoutException {
+            return src.get(timeout, unit);
+        }
+
+        @Override
+        public void run() {
+            src.run();
+        }
+    }
+
+    public static class PriorityFutureComparator implements Comparator<Runnable> {
+        public int compare(Runnable r1, Runnable r2) {
+            if (r1 == null && r2 == null) {
+                return 0;
+            } else if (r1 == null) {
+                return -1;
+            } else if (r2 == null) {
+                return 1;
+            }
+            int p1 = ((PriorityFuture<?>) r1).getPriority();
+            int p2 = ((PriorityFuture<?>) r2).getPriority();
+            return Integer.compare(p1, p2);
```

Comment:
if HIGHEST priority is actually the lowest integer - 1 - wouldn't this method return the reverse of what we expect - compare(HIGHEST, LOWEST) would return -1 as HIGHEST is a lower int

Reply:
added tests